### PR TITLE
feat: added likes and watch-later

### PR DESCRIPTION
**Pull request by Hardiegogo**
- added liked videos page
- added watch later videos page
- added routes to these pages 
- added NavLinks in the side bar
- added RequireAuth for pages that require authentication
- created utility functions for addition and removal of videos to likes/watchlater

screenshots:

image

image

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,7 +3,10 @@ import "./App.css";
 import Login from "./pages/auth-pages/Login";
 import Signup from "./pages/auth-pages/Signup";
 import Homepage from "./pages/home-page/Homepage";
+import LikedVideos from "./pages/liked-video-page/LikedVideos";
 import Videolisting from "./pages/video-listing-page/Videolisting";
+import WatchLater from "./pages/watch-later-page/WatchLater";
```
Comment (aman11s) [inline src/App.js:8]: You can use barrel export here

File: src/components/sidebar-component/Sidebar.jsx [hunk 2]
```diff
@@ -3,17 +3,18 @@ import { Box, ListItem, UnorderedList,Text } from '@chakra-ui/react'
...
-            <ListItem display="flex" alignItems="center" p={2} fontSize="xl" cursor="pointer" pl={10}_hover={{bg:"#0a1924",}} mt={2}><AiFillLike/><Text  ml={1}>Liked Videos</Text></ListItem>
+          <NavLink to='/'><ListItem display="flex" alignItems="center" p={2}  fontSize="xl" cursor="pointer"  pl={10}_hover={{bg:"#0a1924",}}><AiFillHome/><Text ml={1}>Home</Text></ListItem></NavLink>          
+          <NavLink to='/videos'><ListItem display="flex" alignItems="center" p={2} fontSize="xl" cursor="pointer" pl={10}_hover={{bg:"#0a1924",}} mt={2}><BsFillCameraVideoFill/><Text  ml={1}>Videos</Text></ListItem></NavLink>
+          {/* <NavLink><ListItem display="flex" alignItems="center" p={2} fontSize="xl" cursor="pointer" pl={10}_hover={{bg:"#0a1924",}} mt={2}><AiOutlineHistory/><Text  ml={1}>History</Text></ListItem></NavLink>
+          <NavLink><ListItem display="flex" alignItems="center" p={2} fontSize="xl" cursor="pointer" pl={10}_hover={{bg:"#0a1924",}} mt={2}><MdPlaylistPlay/><Text  ml={1}>Playlists</Text></ListItem></NavLink> */}
```
Comment (aman11s) [inline src/components/sidebar-component/Sidebar.jsx:15]: Should avoid commented line of code

**Review by aman11s — APPROVED** [review]
Features are working as intended.
UI looks great 
Few nitpick in code other everything looks fine.

File: src/App.js [hunk 3]
```diff
@@ -14,6 +17,8 @@ function App() {
         <Route path='/videos' element={<Videolisting/>}/>
```
Comment (theVivek-kumar) [inline src/App.js:17]: use outlet  <Outlet> enables the Users component to render its child route

File: src/components/nav-menu-component/NavMenu.jsx [hunk 4]
```diff
@@ -3,11 +3,13 @@ import { Box, ListItem, UnorderedList, Text } from "@chakra-ui/react";
 import { AiFillHome, AiOutlineHistory, AiFillLike } from "react-icons/ai";
 import { BsFillCameraVideoFill } from "react-icons/bs";
 import { MdPlaylistPlay, MdWatchLater } from "react-icons/md";
+import { NavLink } from "react-router-dom";
 
 const NavMenu = () => {
   return (
     <Box position="absolute" top="4.5rem" left="2rem">
       <UnorderedList listStyleType="none" bg="#0a1924" p={8}>
```
Comment (theVivek-kumar) [inline src/components/nav-menu-component/NavMenu.jsx]: use color name

File: src/components/sidebar-component/Sidebar.jsx (hunk 2 above)
Comment (theVivek-kumar) [inline src/components/sidebar-component/Sidebar.jsx:15]: remove commented code

File: src/components/video-card-component/VideoCard.jsx [hunk 5]
```diff
@@ -1,17 +1,55 @@
 import { Text,Box, AspectRatio,Image, Heading } from '@chakra-ui/react'
-import React from 'react'
+import React, { useState } from 'react'
 import { CheckCircleIcon } from '@chakra-ui/icons'
-const VideoCard=({video})=> {
+import VideoOptions from '../video-card-options-component/VideoOptions'
+import {BsThreeDotsVertical} from 'react-icons/bs'
+import { useOutsideClick } from '@chakra-ui/react'
```
Comment (theVivek-kumar) [inline src/components/video-card-component/VideoCard.jsx:6]: just a suggestion use extra icon .js for importing icon

File: src/pages/liked-video-page/LikedVideos.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import React, { useEffect } from "react";
+import {
+  Box,
+  Grid,
+  Heading,
+
+} from "@chakra-ui/react";
+import Navbar from "../../components/navbar-component/Navbar";
```
Comment (theVivek-kumar) [inline src/pages/liked-video-page/LikedVideos.jsx:8]: please do not mix name export with default export

File: src/pages/liked-video-page/LikedVideos.jsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    loadLikes(dispatchVid);
+  }, []);
+
+  const { likedVideos } = vidState;
+
+  return (
+    <Grid
+      bg="bgColor"
+      height="100%"
+      minHeight="100vh"
```
Comment (theVivek-kumar) [inline src/pages/liked-video-page/LikedVideos.jsx:29]: code is good and clean

File: src/utils/video-services/loadWatchLater.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  const encodedToken=localStorage.getItem('token')
+  dispatchVid({type:"VIDEOS_LOADING"})
+    try {
+        const res=await axios({
+            method:"GET",
+            url:"/api/user/watchlater",
+            headers:{authorization:encodedToken}
+        })
+        if(res.status===200){
+            dispatchVid({type:"SET_WATCH_LATER",payload:res.data.watchlater})
```
Comment (theVivek-kumar) [inline src/utils/video-services/loadWatchLater.js:13]: Features are working as intended.
UI looks great and if would look amazing if you give a different color to aside bar and Nav bar slightly color difference

**Review by theVivek-kumar — COMMENTED** [review]
UI is good , good to merge

**Review by spraveenofficial — APPROVED** [review]

**Pull request merged**